src/pages/HomePage.js

```diff
@@ -41,6 +48,15 @@ export const HomePage = () => {
         document.getElementsByTagName("html")[0].style.scrollSnapType = "y proximity";
     }, [searchResultsRef, showSearchResults]);
 
+    useEffect(() => {
+        if (openPasswordRecoveryModal && token) {
+            history.replace({ pathname: "/" });
+            dispatch(toggleAuthModal(2));
```

Comment:
This `2` can cause confusion being left as is (I had to look at the definition of the action function to understand it was a page number). Maybe change the action function name or turn this into a constant.